SEACells/genescores.py

```diff
@@ -50,130 +34,157 @@ def prepare_multiome_anndata(atac_ad, rna_ad, SEACell_label='SEACell', n_bins_fo
-   
-    # RNA - Summarize by metacells
-    rna_metacells = rna_mod_ad.obs[SEACell_label].astype(str).unique()
-    rna_metacells = rna_metacells[rna_mod_ad.obs[SEACell_label].value_counts()[rna_metacells] > 1]
+    # Metacell set
+    rna_metacells = rna_mod_ad.obs[SEACells_label].astype(str).unique()
+    rna_metacells = rna_metacells[rna_mod_ad.obs[SEACells_label].value_counts()[rna_metacells] > 1]
 
+    # Mapping beetween RNA and ATAC
     mapping = mapping.loc[rna_metacells]
-    
     # Create common metacell name
     mapping['common'] = np.arange(len(mapping))
     mapping['common'] = 'metacell ' + mapping['common'].astype(str)
     
-    # Summary matrix 
+    print('Generating Metacell matrices...')
+    print(' RNA')
+   
+    # RNA - Summary matrix 
     summ_matrix = pd.DataFrame(0.0, index=mapping['common'], columns=rna_mod_ad.var_names)
     for m in tqdm(summ_matrix.index):
         rna_meta = mapping.index[mapping['common'] == m][0]
         
-        cells = rna_mod_ad.obs_names[rna_mod_ad.obs[SEACell_label] == rna_meta]
+        cells = rna_mod_ad.obs_names[rna_mod_ad.obs[SEACells_label] == rna_meta]
         summ_matrix.loc[m, :] = np.ravel(rna_mod_ad[cells, :].X.sum(axis=0))
 
     # RNA - create metacell matrix
     rna_meta_ad = _create_ad(summ_matrix)
     rna_meta_ad.obs['original_rna'] = rna_metacells
+
+    # RNA- Normalize 
+    _normalize_ad(rna_meta_ad)
     
-    print(' ATAC')
 
+
+    print(' ATAC')
     # ATAC - Summarize by metacells
     # Summary matrix
     summ_matrix = pd.DataFrame(0.0, index=mapping['common'], columns=atac_mod_ad.var_names)
+    svd = pd.DataFrame(atac_mod_ad.obsm['X_svd'], index=atac_mod_ad.obs_names)
+    meta_svd = pd.DataFrame(0.0, index=mapping['common'], columns=svd.columns)
     for m in tqdm(summ_matrix.index):
         atac_metacell = mapping.loc[mapping['common']== m, 'atac'].item()
-        cells = atac_mod_ad.obs_names[atac_mod_ad.obs[SEACell_label] == atac_metacell]
-        summ_matrix.loc[m, :] = np.ravel(atac_mod_ad[cells, :].layers['TFIDF'].sum(axis=0))
+        cells = atac_mod_ad.obs_names[atac_mod_ad.obs[SEACells_label] == atac_metacell]
+        summ_matrix.loc[m, :] = np.ravel(atac_mod_ad[cells, :].X.sum(axis=0))
+        meta_svd.loc[m, :] = svd.loc[cells, :].sum(axis=0)
 
     # ATAC - create metacell anndata
-    atac_meta_ad = _create_ad(summ_matrix, atac_mod_ad, n_bins_for_gc)
+    atac_meta_ad = _create_ad(summ_matrix)
+    atac_meta_ad.obsm['X_svd'] = meta_svd
     atac_meta_ad.obs['original_atac'] = mapping['atac'].values
     
-    # Add summarized X_svd representation
-    svd = pd.DataFrame(atac_mod_ad.obsm['X_svd'], index=atac_mod_ad.obs_names)
-    summ_svd = svd.groupby(atac_mod_ad.obs[SEACell_label]).mean()
-    atac_meta_ad.obsm['X_svd'] = summ_svd.loc[atac_meta_ad.obs['original_atac'], :].values
+    # ATAC - Normalize
+    sc.pp.filter_genes(atac_meta_ad, min_cells=1)
+    _normalize_ad(atac_meta_ad)
+    _add_atac_meta_data(atac_meta_ad)
```

Comment:
Just need to pass through params:
`_add_atac_meta_data(atac_meta_ad, atac_mod_ad, n_bins_for_gc)`